apps/site/lib/site_web/templates/fare/_nearby_locations.html.eex

```diff
@@ -1,16 +1,48 @@
-<div class="fare-sales-locations">
-  <h3>Nearby MBTA fare retail sales locations</h3>
-  <div class="c-location-cards c-location-cards--background-gray small-set grouped">
-    <%= for {%Fares.RetailLocations.Location{name: name, address: address} = location, distance} <- @fare_sales_locations do %>
-      <%= link to: GoogleMaps.direction_map_url(@search_position, location), class: "c-location-card c-location-card--flex", target: "_blank" do %>
-        <div class="c-location-card__name">
-          <%= name %>
+<div class="c-sales-locations__cards">
+  <%= for {location, distance} <- @locations do %>
+    <%
+      name = Map.get(location, :name)
+      address = Map.get(location, :address)
+      retail_fvm = Map.get(location, :retail_fvm)
+      stop_id = Map.get(location, :stop_id)
+    %>
+    <div class="c-sales-locations__card">
+      <div class="c-sales-locations__card-flex-row">
+        <div class="c-sales-locations__card-place">
+          <%= if stop_id do %>
+            <%= link name, to: stop_path(@conn, :show, stop_id) %>
+          <% else %>
+            <%= name %>
+          <% end %>
         </div>
-        <div class="c-location-card__distance"><%= round_distance(distance) %></div>
-        <div class="c-location-card__description">
+        <div class="c-sales-locations__card-distance"><%= round_distance(distance) %></div>
+      </div>
+      <%= if address do %>
+        <div class="c-sales-locations__card-location">
           <%= address %>
         </div>
       <% end %>
-    <% end %>
-  </div>
+      <%= if retail_fvm do %>
+        <div class="c-sales-locations__card-type">
+          <%= retail_fvm %>
+        </div>
+      <% end %>
+
+      <div class="dropdown show c-sales-locations__card-directions">
+        <% url_address = if address, do: address, else: "#{location.latitude},#{location.longitude}" %>
```

Comment:
Is there a scenario in which `url_address` is `address`? I tried different locations and it was always `latitude, longitude` 🤷‍♀️

Reply:
For PSLs, I think it'll always be address

Reply:
Gotcha! Thanks, I see it now.